model/vfs/vfsafero/fsck.go

```diff
@@ -135,13 +151,19 @@ func (afs *aferoVFS) checkFiles(entries map[string]*vfs.TreeFile, accumulate fun
 						MD5SumIndex: f.MD5Sum,
 					},
 				})
+				if failFast {
+					return errFailFast
+				}
 			}
 		}
 		delete(entries, fullpath)
 		return nil
 	})
 	if err != nil {
-		return
+		if err == errFailFast {
+			return nil
+		}
+		return err
```

Comment:
I don't understand all these `if failfast return nil`. Wouldn't this hide errors and a risk to return that everything was ok ?

Reply:
We have two things: errors and fsckLogs. Errors are for when the results are unknown (CouchDB not available for example), and fsckLogs are the messages displayed by the FSCK. When a FSCK log is found, the stack will first send it, and if the `--fail-fast` option has been used, it will stop the FSCK. To do that, we have several functions and the simpler way is to return a special error (`errFailFast`), and to transform it to nil later to make the HTTP response a 200 OK, and not a 5xx.